frontend/packages/knative-plugin/src/utils/fetch-dynamic-eventsources-utils.ts

```diff
@@ -76,7 +77,14 @@ export const fetchEventSourcesCrd = async () => {
             namespaced: true,
             crd: true,
             color: knativeEventingColor.value,
-          });
+          };
+          const sourceIndex = _.findIndex(accumulator, ['kind', kind]);
+          // added check as some sources has multiple entries with deprecated APIgroups
+          if (sourceIndex === -1) {
+            accumulator.push(sourceModel);
+          } else if (!labels?.['eventing.knative.dev/deprecated'] === true) {
```

Comment:
Why not just add this check to the `if (version)` check above? Not sure why we need to first push it to accumulator and then remove it.

Reply:
the reason is can't just filter on `labels?.['eventing.knative.dev/deprecated']` as for some sources we get two entries with different apiGroups (one will be deprecated) i.e for `ApiServerSources/SinkBinding`. Second scenario is for `ConatinerSource` we just get one entry and that has deprecated label. So we needed to have list of sources with no deprecated apiGroup otherwise deprecated one